File: 008.StringToInteger/wyrj/StringtoInteger.js
/* 208ms */
/**
 * @param {string} str
 * @return {number}
 */
var myAtoi = function(str) {
    var INT_MAX = 2147483647, INT_MIN = -2147483648;
    str = str.trimLeft();
    var j = 1;
    while('0' <= str[j] && '9' >= str[j]) {
        j++;
    }
    var ret = parseInt(str.slice(0, j), 10);
    return (isNaN(ret)) ? 0 : (ret > INT_MAX) ? INT_MAX : (ret < INT_MIN) ? INT_MIN : ret;
};

File: 008.StringToInteger/wyrj/StringtoInteger.c
/* 4ms */
int myAtoi(char* str) {
    int start = 0;
    while(' ' == str[start]) {
        start++;
    }
    int ret = 0, INTMAX = 2147483647, INTMIN = -2147483648, index, n;
    if ('-' == str[start]) {
        index = start + 1;
        while('0' <= str[index] && '9' >= str[index]) {
            n = str[index++] - '0';
            if ((INTMIN + n) / 10 > ret) {
                return INTMIN;
            }
            ret = ret * 10 - n;
        }
    } else {
        index = ('+' == str[start]) ? start + 1 : start;
        while('0' <= str[index] && '9' >= str[index]) {
            n = str[index++] - '0';
            if ((INTMAX - n) / 10 < ret) {
                return INTMAX;
            }
            ret = ret * 10 + n;
        }
    }
    return ret;
}